platform/ecsv3/ecs.go

```diff
@@ -55,10 +59,7 @@ func NewECSPlatform(ctx context.Context, metadataURI string, executionEnv string
 		return nil, err
 	}
 
-	meta, err := c.GetTaskMetadata(ctx)
-	if err != nil {
-		return nil, err
-	}
+	meta := getTaskMetadata(ctx, c, taskMetadataWaitForReadyInterval)
```

Comment:
`getTaskMetadata` will return `nil` if `ctx` is expired but next `detectNetworkMode` refers fields of `meta`. Therefore runtime will panic.

I think `getTaskMetadata` had better returns an error.

Reply:
That is as you pointed out.
I will fix to return an error when the context is canceled.